modules/oe_editorial_corporate_workflow/modules/oe_editorial_corporate_workflow_translation/tests/src/Functional/CorporateWorkflowTranslationRevisionTest.php

```diff
@@ -549,4 +552,95 @@ public function testTranslationRevisionsCarryOver(): void {
     $this->assertEquals('the paragraph text value FR 2', $paragraph_translation->get('field_workflow_paragraph_text')->value);
   }
 
+  /**
+   * Tests that moderation state translations are kept in sync with original.
+   */
+  public function testModerationStateSync(): void {
+    // Create a validated node and add a translation.
+    /** @var \Drupal\node\NodeStorageInterface $node_storage */
+    $node_storage = $this->entityTypeManager->getStorage('node');
+
+    /** @var \Drupal\node\NodeInterface $node */
+    $node = $node_storage->create([
+      'type' => 'page',
+      'title' => 'My node',
+      'moderation_state' => 'draft',
+    ]);
+    $node->save();
+    $node = $this->moderateNode($node, 'validated');
+    $this->drupalGet(Url::fromRoute('oe_translation.permission_translator.create_local_task', [
+      'entity' => $node->id(),
+      'source' => 'en',
+      'target' => 'fr',
+      'entity_type' => 'node',
+    ]));
+    $this->assertSession()->elementContains('css', '#edit-title0value-translation', 'My node');
+    $values = [
+      'Translation' => 'My node FR',
+    ];
+    // It should be the first local task item created so we use the ID 1.
+    $url = Url::fromRoute('entity.tmgmt_local_task_item.canonical', ['tmgmt_local_task_item' => 1]);
+    $this->drupalGet($url);
+    $this->submitForm($values, t('Save and complete translation'));
+
+    // Assert that the node has two translations and the moderation state entity
+    // also has two translations.
+    $node_storage->resetCache();
+    /** @var \Drupal\node\NodeInterface $node */
```

Comment:
This is redundant. You can remove it.

Reply:
No it is not...entity storage returns EntityInterface, so I just type hinted.